platform-hub-api/app/services/announcements_processor_service.rb

```diff
@@ -15,16 +15,22 @@ def process announcement
     @logger.info "Processing announcement #{announcement.id} - #{announcement.level} - '#{announcement.title}'"
 
     begin
-      announcement.update! status: :delivering
 
-      unless announcement.deliver_to.blank?
-        process_for_email_delivery announcement
-        process_for_slack_delivery announcement
+      announcement.with_lock do
```

Comment:
I'll rollback this particular change until we've had a chance to discuss it properly.